src/main/java/seedu/address/logic/commands/MarkPaidTarget.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.logic.commands;
+
+import seedu.address.commons.core.index.Index;
+
+/**
+ * Represents the target of the MarkPaidCommand, which can be either an Index or 'all'.
+ */
+public class MarkPaidTarget {
+    private final Index index;
+    private final boolean markAll;
+
+    private MarkPaidTarget(Index index, boolean markAll) {
+        this.index = index;
+        this.markAll = markAll;
+    }
+
+    public static MarkPaidTarget of(Index index) {
```

Comment:
`of` is too undescriptive
Suggest: `fromIndex`

Reply:
Renamed